network-area-diagram-viewer/src/main/resources/svg.js

```diff
@@ -0,0 +1,463 @@
+
+// TODO(Luma) consider three winding transformers
+// TODO(Luma) consider nodes with multiple buses
+// TODO(Luma) separate components of edges, having one that can be "scaled" (length line from exit of node to middle point)
+//              for reference, see Android NinePatch drawables
+//              https://developer.android.com/develop/ui/views/graphics/drawables#nine-patch
+// TODO(Luma) JavaFX seems to have problems drawing a translated foreign object, try to enclose it in a <g> element?
+// TODO(Luma) update on client: find related hidden nodes and update initial position in metadata?
+
+window.addEventListener('load', function() {
+    var svgDiagram = document.getElementsByTagName("svg")[0];
+    if (svgDiagram != null) {
+        makeDraggableSvg(svgDiagram);
+    }
+});
+
+// Draggable SVG Elements from
+// https://www.petercollingridge.co.uk/tutorials/svg/interactive/dragging/
+function makeDraggableSvg(svg) {
+    var selectedElement = false, offset, transform;
+    var diagram = new Diagram(svg);
+
+    svg.addEventListener('mousedown', startDrag);
```

Comment:
```suggestion
    svg.getElementsByClassName("nad-vl-nodes")[0].addEventListener('mousedown', startDrag);
```

Reply:
Now we also have boundary nodes (that are not voltage level nodes) representing the ends of dangling lines.
And in the future we may have other elements in the diagram that are draggable (the drawing for the 3wt nodes).
I prefer to capture the event in the whole svg and then check which element (if any) should be draggable in a separate method.
This also keeps the methods related to `makeDraggable` less coupled with the actual contents of the SVG. I am trying to put all the knowledge about the inners of the diagram inside the `Diagram` class of the JavaScript.